features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/TextSemanticsNodeMapper.kt

```diff
@@ -0,0 +1,148 @@
+internal class TextSemanticsNodeMapper(colorStringFormatter: ColorStringFormatter) :
+    AbstractSemanticsNodeMapper(colorStringFormatter) {
+    override fun map(semanticsNode: SemanticsNode, parentContext: UiContext): ComposeWireframe {
+        val text = resolveText(semanticsNode.config)
+        val textStyle = resolveTextStyle(semanticsNode, parentContext) ?: defaultTextStyle
+        val bounds = resolveBound(semanticsNode)
+        return ComposeWireframe(
+            MobileSegment.Wireframe.TextWireframe(
+                id = semanticsNode.id.toLong(),
+                x = bounds.x,
+                y = bounds.y,
+                width = bounds.width,
+                height = bounds.height,
+                text = text ?: "",
+                textStyle = textStyle,
+                textPosition = resolveTextAlign(semanticsNode)
+            ),
+            null
+        )
+    }
+
+    private fun resolveTextAlign(semanticsNode: SemanticsNode): MobileSegment.TextPosition? {
+        return resolveSemanticsTextStyle(semanticsNode)?.let {
+            val align = when (it.textAlign) {
+                TextAlign.Start,
+                TextAlign.Left -> MobileSegment.Horizontal.LEFT
+
+                TextAlign.End,
+                TextAlign.Right -> MobileSegment.Horizontal.RIGHT
+
+                TextAlign.Justify,
+                TextAlign.Center -> MobileSegment.Horizontal.CENTER
+
+                else -> MobileSegment.Horizontal.LEFT
+            }
+            MobileSegment.TextPosition(
+                alignment = MobileSegment.Alignment(
+                    horizontal = align
+                )
+            )
+        }
+    }
+
+    private fun resolveTextStyle(semanticsNode: SemanticsNode, parentContext: UiContext): MobileSegment.TextStyle? {
+        return resolveSemanticsTextStyle(semanticsNode)?.let { textStyle ->
+            val color = resolveModifierColor(semanticsNode) ?: textStyle.color
+            MobileSegment.TextStyle(
+                family = when (val value = textStyle.fontFamily) {
+                    is GenericFontFamily -> value.name
+                    else -> DEFAULT_FONT_FAMILY
+                },
+                size = textStyle.fontSize.value.toLong(),
+                color = convertColor(color.value.toLong()) ?: parentContext.parentContentColor
+                    ?: DEFAULT_TEXT_COLOR
+            )
+        }
+    }
+
+    private fun resolveModifierColor(semanticsNode: SemanticsNode): Color? {
+        val modifier = semanticsNode.layoutInfo.getModifierInfo().firstOrNull {
+            ComposeReflection.TextStringSimpleElement?.isInstance(it.modifier) ?: false
+        }?.modifier
+        modifier?.let {
+            if (ComposeReflection.TextStringSimpleElement?.isInstance(it) == true) {
+                val colorProducer = ComposeReflection.ColorProducerField?.getSafe(it) as? ColorProducer
+                return colorProducer?.invoke()
+            }
+        }
+        return null
```

Comment:
seems can be just `return if (modifier != null && ComposeReflection.TextStringSimpleElement?.isInstance(modifier) == true) { .... } else null`